[20.03] backport: nixos/btrfs autoScrub: don't fail when scrub finishes successfully

symphorien opened the pull request:
(cherry picked from commit 55d16d533446a815c5d266b72ff84bc97300fddc)

I tested that a scrub can finish successfully and be cancelled successfully.

symphorien committed:
nixos/btrfs autoScrub: don't fail when scrub finishes successfully

(cherry picked from commit 55d16d533446a815c5d266b72ff84bc97300fddc)

symphorien commented:
cc @timokau who merged #84135

timokau commented:
Have you only tried this once or have you been using it for a while? The testing standard for stable-branches is a bit higher than unstable, we should be reasonably sure that this will not break anybodies setup. Given that, do you think its good to go?

symphorien commented:
When I opened the PR, I switched my system to this branch, and successfully ran on scrub and cancelled one. Then I switched my system to stable again.

timokau commented:
A one-time test might not catch unintended longterm consequences or breakage in edge-cases. In that case I'd rather let this sit in unstable for a bit. If nothing has happened until then, please ping me again in a week. Or if you're actively testing it yourself, 3 days are probably enough too.

symphorien commented:
cc @timokau

timokau commented:
Thanks!
